internal/services/oracle/oracle_database_system_versions_data_source.go

```diff
@@ -0,0 +1,180 @@
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+)
+
+type DatabaseVersionsDataSource struct{}
+
+type DatabaseVersionsModel struct {
+	Location                       string                     `tfschema:"location"`
+	DatabaseSystemShape            string                     `tfschema:"database_system_shape"`
+	DatabaseSoftwareImageSupported *bool                      `tfschema:"database_software_image_supported"`
+	ShapeFamily                    string                     `tfschema:"shape_family"`
+	StorageManagement              string                     `tfschema:"storage_management"`
+	UpgradeSupported               *bool                      `tfschema:"upgrade_supported"`
+	Versions                       []DatabaseVersionItemModel `tfschema:"versions"`
+}
+
+type DatabaseVersionItemModel struct {
+	Name                  string `tfschema:"name"`
+	LatestForMajorVersion bool   `tfschema:"latest_for_major_version"`
+	SupportsPdb           bool   `tfschema:"supports_pdb"`
+	Version               string `tfschema:"version"`
+}
+
+func (d DatabaseVersionsDataSource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"location": commonschema.Location(),
+
+		// Optional filters
+
+		"database_system_shape": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			ValidateFunc: validation.StringInSlice(dbversions.PossibleValuesForBaseDbSystemShapes(), false),
+		},
+		"database_software_image_supported": {
+			Type:        pluginsdk.TypeBool,
+			Optional:    true,
+			Description: "Whether to filter versions that support database software images.",
+		},
+		"shape_family": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			ValidateFunc: validation.StringInSlice(dbversions.PossibleValuesForShapeFamilyType(), false),
+			Description:  "The shape family of the DB System to filter versions by.",
+		},
+		"storage_management": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			ValidateFunc: validation.StringInSlice(dbversions.PossibleValuesForStorageManagementType(), false),
+			Description:  "The storage management type to filter versions by.",
+		},
+		"upgrade_supported": {
+			Type:        pluginsdk.TypeBool,
+			Optional:    true,
+			Description: "Whether to filter versions that support upgrades.",
+		},
+	}
+}
+
+func (d DatabaseVersionsDataSource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"versions": {
+			Type:        pluginsdk.TypeList,
+			Computed:    true,
+			Description: "A list of available Oracle Database versions and their properties.",
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"name": {
+						Type:     pluginsdk.TypeString,
+						Computed: true,
+					},
+					"latest_for_major_version": {
+						Type:     pluginsdk.TypeBool,
+						Computed: true,
+					},
+					"supports_pdb": {
```

Comment:
the name of a `bool` type should be ended with `enabled`, ([ref](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/topics/schema-design-considerations.md#features-that-are-toggled-by-the-property-enabled))

Reply:
not resolved?